controllers/replica_set_controller.go

```diff
@@ -224,7 +224,8 @@ func (r ReplicaSetReconciler) Reconcile(ctx context.Context, request reconcile.R
 			withMongoDBMembers(mdb.AutomationConfigMembersThisReconciliation()).
 			withStatefulSetReplicas(mdb.StatefulSetReplicasThisReconciliation()).
 			withMessage(None, "").
-			withRunningPhase(),
+			withRunningPhase().
+			withVersion(&mdb),
```

Comment:
related to the other comment, I think `withVersion(mdb.GetMongoDBVersion())` would be much clearer